baseplate/core.py

```diff
@@ -83,15 +84,18 @@ def new(cls):
 
         """
         trace_id = random.getrandbits(64)
-        return cls(trace_id=trace_id, parent_id=None, span_id=trace_id)
+        return cls(trace_id=trace_id, parent_id=None,
+                   span_id=trace_id, sampled=None, flags=0)
 
     @classmethod
-    def from_upstream(cls, trace_id, parent_id, span_id):
+    def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
         """Build a TraceInfo from individual headers.
 
         :param int trace_id: The ID of the trace.
         :param int parent_id: The ID of the parent span.
         :param int span_id: The ID of this span within the tree.
+        :param str sampled: Boolean flag of value "0" or "1" to determine request sampling.
```

Comment:
can this be an actual true/false boolean in python rather than being stringly typed?